blog/content/post/2025-08-12-ram-wipe-further-analysis.md

```diff
@@ -0,0 +1,594 @@
+our prior ram-wipe experiments to rigorously evaluate if `init_on_free` can
+serve as a robust safeguard, perhaps supplanting existing, less comprehensive
+memory wiping solutions.'
+cover: /covers/ram-wipe.png
+author: kamil.aronowski
+layout: post
+published: true
+date: 2025-08-12
+archives: "2025"
+
+tags:
+  - attack
+  - dasharo
+  - debian
+  - encryption
+  - firmware
+  - linux
+  - luks
+  - open-source
+  - qemu
+  - ram
+  - uefi
+  - warm
+  - wipe
+categories:
+  - Firmware
+  - Security
+
+---
+
+This post will follow up on our journey with the secure [ram-wipe
+solution](https://github.com/kicksecure/ram-wipe). After [diving deep into its
+capabilities recently](https://blog.3mdeb.com/2025/2025-05-20-ram-wipe/), we
+are excited to share our heartfelt discoveries of how [our modified, simplified
+version](https://github.com/zarhus/ram-wipe/tree/88091fea2c58c7f1345c4a05e92fddfecbe1d807)
+fiercely protects your electronics from unseen RAM-based attacks. Join us in
+unveiling this essential defense that could change the future of operational
+security.
+
+This research has been supported by [Power Up
+Privacy](https://powerupprivacy.com/), a privacy advocacy group that seeks to
+provide support to privacy projects so they can accomplish their objective of
+improving the world.
+
+## Introduction
+
+The `init_on_free` boot option enables a mechanism in Linux that zeroes out
+memory after it has been freed. According to [its relevant thread in the Linux
+Kernel Mailing
+List](https://lore.kernel.org/all/20190617151050.92663-1-glider@google.com/),
+
+> Enabling init_on_free also guarantees that pages and heap objects are
+> initialized right after they're freed, so it won't be possible to access
+> stale data by using a dangling pointer.
+>
+> [...]
+>
+> The init_on_free feature is regularly requested
+> by folks where memory forensics is included in their thread models.
+
+A good approach to operational security involves the knowledge of how exactly
+one's tools operate and what their limitations are. It would be disastrous to
+expect a workstation with anti-memory forensics mechanisms to fail at the most
+unexpected time and place. As the [Kicksecure documentation on
+ram-wipe](https://www.kicksecure.com/wiki/Dev/RAM_Wipe#ram-wipe_improvements)
+suggests, several ram-wipe components might require improvements.
+
+Today, we will explore an alternative: whether it's sufficient to replace the
+aforementioned troublesome mechanisms with the kernel-supplied `init_on_free`
+option to protect against user space information disclosure.
+
+## Testing methodology
+
+Just like in [our earlier
+exploration](https://blog.3mdeb.com/2025/2025-05-20-ram-wipe/), we will
+experiment with the same [EFI
+application](https://github.com/zarhus/ram-dump-efi) for determining the
+locations of EFI structures and dumping the memory after a system reboot.
+Furthermore, we will use the [Dasharo firmware](https://www.dasharo.com/),
```

Comment:
I would link to a specific version of Dasharo used. dasharo.com is a terrible target; it was not updated for a long time, and we always have excuses not to fix that. Better is docs.dasharo.com, but the best is the overview page of the target platform we are using.